tests/src/test/java/alluxio/job/master/JobMasterIntegrationTest.java

```diff
@@ -142,4 +148,14 @@ public void getAllWorkerHealth() throws Exception {
     assertNotNull(workerHealth.getHostname());
     assertEquals(3, workerHealth.getLoadAverage().size());
   }
+
+  @Test
+  public void testThreadPoolSize() throws Exception {
+    mJobMaster.setTaskPoolSize(0);
+    long jobId = mJobMaster.run(new SleepJobConfig(1));
+    CommonUtils.sleepMs(300);
```

Comment:
why is the sleep necessary?

Reply:
The sleep is to show that this job never completes despite the waiting.

Reply:
Is there a way to not depend on sleeps for testing? I would like for integration tests to complete in less time...